templates/glanceapi/config/glance-api-config.json

```diff
@@ -0,0 +1,17 @@
+{
+    "command": "/usr/bin/glance-api --config-file /etc/glance/glance-api.conf",
```

Comment:
should add --config-dir=/etc/glance/glance.d and then have custom.conf?
```
        {
            "source": "/var/lib/config-data/merged/custom.conf",
            "dest": "/etc/glance/glance.d/custom.conf",
            "owner": "glance",
            "perm": "0600"
        },
```
https://docs.openstack.org/glance/latest/configuration/configuring.html

Reply:
Somehow I missed copying this pattern from placement.  Will add.

Reply:
yes thats how we used it in cnosp. there was one concern in a meeting some week ago. some openstack services support having multiple same parameters in the conf. nova is one of them. Therefore I changed how we used it there to just merge the required parameters into the main service config and conf snipped provided via CustomServiceConfig into the conf.d of the service. if the service supports multiple same parameters you could do that with that. 

Now I see "If --config-dir is set, then --config-file is ignored." , but it looks like it supports using multiple --config-file parameters. Have you tried that using `/usr/bin/glance-api --config-file /etc/glance/glance-api.conf  --config-file /etc/glance/glance.conf.d/custom.conf` ?

Reply:
That does work, but I'm not sure how it respects order for overwrites.  Back to your other thought though, re: `conf-dir`, we could just put `00-glance-api.conf` and `01-custom.conf` in `glance.conf.d` dir and use the numbering to ensure processing order.  Thoughts?

Reply:
yes that would work. could then not merge at all and just add the required to 00 and customization to 01 where the service supports a conf dir, but then we should do that for all components where a conf dir is supported to have the same behavior.

re the ordering, from `podman run -it -u root --rm quay.io/tripleotraincentos8/centos-binary-glance-api:current-tripleo glance-api --help` -> "Values in latter files take precedence"

```
  --config-file PATH    Path to a config file to use. Multiple config files
                        can be specified, with values in later files taking
                        precedence. Defaults to None. This option must be set
                        from the command-line.
```

Reply:
ack sounds good. maybe we see what works for all/most of the services and do the same everywhere.